images/cilium/tests/main.tf

```diff
@@ -16,36 +24,38 @@ data "oci_exec_test" "operator-version" {
   digest = var.digests.operator
 }
 
-# TODO:
-# To run this test, we need to create a cluster that looks like this
-#
-# k3d cluster create -i cgr.dev/chainguard/k3s:latest \
-#     --k3s-arg '--disable=traefik@server:*' \
-#     --k3s-arg '--disable=metrics-server@server:*' \
-#     --k3s-arg '--disable-network-policy@server:*' \
-#     --k3s-arg '--flannel-backend=none@server:*' \
-#
-# for node in k3d-k3s-default-server-0; do
-#   docker exec -i $node /bin/sh <<-EOF
-#       mount bpffs -t bpf /sys/fs/bpf
-#       mount --make-shared /sys/fs/bpf
-#       mkdir -p /run/cilium/cgroupv2
-#       mount -t cgroup2 none /run/cilium/cgroupv2
-#       mount --make-shared /run/cilium/cgroupv2/
-#
-#
-# resource "helm_release" "cilium" {
-#   repository       = "https://helm.cilium.io"
-#   chart            = "cilium"
-#   name             = "cilium"
-#   namespace        = "kube-system"
-
-#   values = [jsonencode({
-#     operator = {
-#       image = {
-#         override = var.digests.operator
-#       }
-#       replicas = 1
-#     }
-#   })]
-# }
+data "oci_exec_test" "create-cluster" {
+  script = "${path.module}/create-cluster.sh"
+  digest = var.digests.operator
+}
+
+resource "helm_release" "cilium" {
+  depends_on = [data.oci_exec_test.create-cluster]
+  repository = "https://helm.cilium.io"
+  chart      = "cilium"
+  name       = "cilium"
+  namespace  = "kube-system"
+
+  values = [jsonencode({
+    operator = {
+      image = {
+        override = var.digests.operator
```

Comment:
do we need the agent image as well? or is that coming later?

Reply:
agent image is coming